src/3id/__tests__/3id.test.js

```diff
@@ -87,6 +89,36 @@ describe('3id', () => {
       expect(opts.consentCallback).toHaveBeenCalledWith(false)
       expect(mockedUtils.openBoxConsent).toHaveBeenCalledTimes(0)
     })
+
+    it('should create a new identity when passed a contentSignature', async () => {
+      const opts = { consentCallback: jest.fn(), contentSignature: '0xsomeContentSignature' }
+      const contentSignatureThreeId = await ThreeId.getIdFromEthAddress(ADDR_3, ETHEREUM, ipfs, opts)
+      expect(contentSignatureThreeId.serializeState()).toEqual(ADDR_3_STATE_1)
+      expect(contentSignatureThreeId.DID).toMatchSnapshot()
+      expect(opts.consentCallback).toHaveBeenCalledWith(true)
+      expect(mockedUtils.openBoxConsent).toHaveBeenCalledTimes(0)
+      expect(await resolve(contentSignatureThreeId.DID)).toMatchSnapshot()
+    })
+
+    it('should create the same identity given the same address and contentSignature', async () => {
+      // did is mocked, so compares serialized state
+      const opts = { contentSignature: '0xsomeContentSignature' }
+      const threeId1 = await ThreeId.getIdFromEthAddress(ADDR_3, ETHEREUM, ipfs, opts)
+      clearLocalStorage3id(ADDR_3)
+      const threeId2 = await ThreeId.getIdFromEthAddress(ADDR_3, ETHEREUM, ipfs, opts)
+      expect(mockedUtils.openBoxConsent).toHaveBeenCalledTimes(0)
+      expect(threeId1.serializeState()).toEqual(threeId2.serializeState())
+    })
+
+    it('should NOT create the same identity given the same address but a different contentSignature', async () => {
+      // did is mocked, so compares serialized state
+      const opts = { contentSignature: '0xsomeIncorrectSignature' }
+      const threeId1 = await ThreeId.getIdFromEthAddress(ADDR_3, ETHEREUM, ipfs, opts)
+      clearLocalStorage3id(ADDR_3)
+      const threeId2 = await ThreeId.getIdFromEthAddress(ADDR_3, ETHEREUM, ipfs, opts)
+      expect(mockedUtils.openBoxConsent).toHaveBeenCalledTimes(0)
+      expect(threeId1.serializeState()).not.toEqual(threeId2.serializeState())
```

Comment:
I'm not really following why this line would be not equal. Isn't the same consentSignature being used for both ThreeId instances? I also don't see any difference in this test and the previous one. What am I missing?

Reply:
The contentSignature on lines 105 and 115 are different:
105: `const opts = { contentSignature: '0xsomeContentSignature' }`
vs.
115: `const opts = { contentSignature: '0xsomeIncorrectSignature' }`

This could be made more clear by using capitalized constants as is done for addresses. I will make that change now.

Reply:
Just pushed c7d8815 which should make this clearer

Reply:
I see that they are different in the two different tests, but in this test uses the opts form line 117 for both `threeId1` and `threeId2` so I don't understand why their serialized states would be different?

Reply:
@danjm Did you see my last comment? I'm still having trouble understanding this. But I'm probably missing something trivial. Will try to run locally today to get a better understanding.

Reply:
Oh I see now! It's because it loads the state from localstorage for `threeId1` and then it's cleared for the second one!

Reply:
Ah, nice catch. I didn't notice that before... and I guess I didn't really understand your initial comment here until now. Looking at this now, I would write the test differently so it wasn't dependent on that local storage clear. I guess this works for now though. Thanks for following up on that.

Reply:
I see. Should be fine for now yes. Just started preparing a release for this so you can get on with your integration: https://github.com/3box/3box-js/pull/540 Should be ready by the end of the day I think.